bigquery/schools_joined_with_metrics.sql

```diff
@@ -0,0 +1,144 @@
+IF
+  (historic_signups.School_been_added IS TRUE,
+    TRUE,
+  IF
+    (school_user_metrics.number_of_users >0,
+      TRUE,
+      FALSE)) AS signed_up, #similarly, if the school had signed up pre-DSI, sets signed_up to true; otherwise, work this out from the number of users the school has on DSI
+  school_user_metrics.number_of_users AS number_of_users,
+  IFNULL(school_vacancy_metrics.vacancies_published,
+    0) AS vacancies_published, #convert null values for vacancies_published into zeros
+  IFNULL(school_vacancy_metrics.vacancies_published_in_the_last_year,
+    0) AS vacancies_published_in_the_last_year,
+  IFNULL(school_vacancy_metrics.vacancies_published_in_the_last_quarter,
+    0) AS vacancies_published_in_the_last_quarter,
+  IFNULL(school_vacancy_metrics.vacancies_published_in_the_last_quarter,
+    0) AS vacancies_currently_live,
+  GIAS.EstablishmentTypeGroup__name_ AS establishment_type_group,
+  GIAS.PhaseOfEducation__name_ AS education_phase,
+  GIAS.ReligiousCharacter__name_ AS religious_character,
+  GIAS.ReligiousEthos__name_ AS religious_ethos,
+  GIAS.NumberOfPupils AS number_of_pupils,
+  GIAS.Trusts__name_ AS trust_name,
+  mat_metrics.trust_size AS academies_in_trust,
+  GIAS.TelephoneNum AS telephone_number,
+  GIAS.HeadFirstName AS head_first_name,
+  GIAS.HeadLastName AS head_last_name,
+  CONCAT(GIAS.HeadFirstName," ",GIAS.HeadLastName) AS head_name, #stick these together for easy use in mailing lists etc. further down the pipeline
+  GIAS.GOR__name_ AS GOR,
+  GIAS.RSCRegion__name_ AS RSC_region,
+IF
+  (GIAS.Trusts__name_ IS NULL,
+    "Single school",
+  IF
+    (mat_metrics.trust_size=1,
+      "SAT",
+      "MAT")) AS tag #categorise the school as either a single school, part of a SAT or part of a MAT to provide a simplified dimension for analysis further down the pipeline
+FROM
+  `teacher-vacancy-service.production_dataset.school` AS school
+LEFT JOIN
+  `teacher-vacancy-service.production_dataset.school_type` AS school_type
+ON
+  school_type.id=school.school_type_id
+LEFT JOIN
+  `teacher-vacancy-service.production_dataset.region` AS region
+ON
+  region.string_field_0=school.region_id
+LEFT JOIN
+  `teacher-vacancy-service.production_dataset.detailed_school_type` AS detailed_school_type
+ON
+  detailed_school_type.id=school.detailed_school_type_id
+LEFT JOIN
+  school_user_metrics
+ON
+  school_user_metrics.urn=school.urn
+LEFT JOIN
+  `teacher-vacancy-service.production_dataset.STATIC_schools_historic_pre201119` AS historic_signups
+ON
+  historic_signups.URN=school.urn
+LEFT JOIN
+  school_vacancy_metrics
+ON
+  school_vacancy_metrics.urn=CAST(school.urn AS STRING)
+LEFT JOIN
+  `teacher-vacancy-service.production_dataset.STATIC_GIAS_Feb20` AS GIAS
```

Comment:
A quick glance at the BigQuery docs *suggests* `DECLARE` *might* work.  Feel free to disregard if you already know it won't or it adds too much work for now. 

https://cloud.google.com/bigquery/docs/reference/standard-sql/scripting#declare